core/src/main/scala/kafka/log/UnifiedLog.scala

```diff
@@ -154,16 +154,41 @@ class UnifiedLog(@volatile var logStartOffset: Long,
   locally {
     initializePartitionMetadata()
     updateLogStartOffset(logStartOffset)
+    updateLocalLogStartOffset(math.max(logStartOffset, localLog.segments.firstSegmentBaseOffset.getOrElse(0L)))
+    if (!remoteLogEnabled())
+      logStartOffset = localLogStartOffset()
```

Comment:
Hmm, logStartOffset could be larger than base offset of the first local segment. So, it seems that we can't just switch to the base offset of the first local segment if remote log is not enabled.

Reply:
We already set the localLogStartOffset as max of passed logStartOffset and the first segment's base offset. 
When remote log is not enabled, `logStartOffset` is set as `localLogStartOffset` as computed above.